AmplifyPlugins/Predictions/AWSPredictionsPlugin/Liveness/Service/WebSocketSession.swift

```diff
@@ -97,5 +108,32 @@ final class WebSocketSession {
         ) {
             onClose(closeCode)
         }
+        
+        // MARK: - URLSessionTaskDelegate methods
+        func urlSession(_ session: URLSession,
+                        task: URLSessionTask,
+                        didFinishCollecting metrics: URLSessionTaskMetrics
+        ) {
+            guard let httpResponse = metrics.transactionMetrics.first?.response as? HTTPURLResponse,
+                  let dateString = httpResponse.value(forHTTPHeaderField: "Date") else {
+                Amplify.log.verbose("\(#function): Couldn't find Date header in URLSession metrics")
+                return
+            }
+            
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "EEE, d MMM yyyy HH:mm:ss z"
+            guard let serverDate = dateFormatter.date(from: dateString) else {
+                Amplify.log.verbose("\(#function): Error parsing Date header in expected format")
```

Comment:
would you want to handle error states in `onServerDateReceived`? i.e. return a null Date or an error state to be handled by the client if there's an issue here?

Reply:
[serverDate](https://github.com/aws-amplify/amplify-swift/pull/3816/files/822d86dbd252a809ee6c49a38272a804d301cdb6#diff-2bd505be1c27be4f9fb9b7d2840fdbb2544a0f26ef6e9bdeb69ad88f368b188eR20:~:text=_%20in%20%7D-,var,-serverDate%3A%20Date) in `WebSocketSession` starts as optional and checked for `nil` 

```
guard 
    exceptionEvent == .invalidSignature,
    connectingState == .normal,
    let savedURLForReconnect = savedURLForReconnect,
    let serverDate = serverDate else {
        onServiceException(.init(event: exceptionEvent))
        return .stopAndInvalidateSession
    }
```
i agree - explicitly setting it as `nil` is a good practice. will update it.